### ISSUE TITLE
Having trouble connecting frontend with the Spring app on Kubernetes

### ISSUE THREAD

**Issue by ivan-rivera** [rinormaloku/k8s-mastery#37]
In the frontend app we used `http://localhost:8080/sentiment` as the endpoint for our sentiment scoring service, but after deploying the app on Kubernetes we need to change this IP to match the web app load balancer service. I have done exactly that and my IP now looks like `http://192.168.64.2:31582/sentiment`. I rebuilt the frontend, built a new docker image and pushed it to Docker Hub. Just to make sure, I also scaled frontend deployment to zero replicas and then back to 2. After doing this, I tried to tet the whole thing, but unfortunately, I got an error:

```
Failed to load resource: net::ERR_CONNECTION_REFUSED localhost:8080/sentiment:1
Uncaught (in promise) TypeError: Failed to fetch 192.168.64.2/:1
```

I'm not entirely sure why the localhost still appears there (my guess is that sa-web-app-deployment.yaml defines SA_LOGIC_API_URL as http://sa-logic which NAT might be translating as localhost), but it looks like it also noticed the new IP I gave it, namely `192.168.64.2`

If I look at the logs of my web app pod, here is what I see (connection refused):

```
2020-07-04 20:30:36.221 ERROR 1 --- [nio-8080-exec-1] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is org.springframework.web.client.ResourceAccessException: I/O error on POST request for "http://localhost/analyse/sentiment": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)] with root cause

```

Any ideas what I could be doing wrong here?

**Comment by rinormaloku** [comment]
Hi Ivan can it be that you have nginx still running from the step of
setting up the application without containers?

**Comment by himanshu85k** [comment]
I am facing same issue. I have also verified that nginx is not running.

For me sa-web-app-lb Loadbalancer service is running on URL 192.168.49.2:31752
![image](https://user-images.githubusercontent.com/33631044/109820771-51bad900-7c5b-11eb-9263-2c123f1a6482.png)

I placed this URL in App.js file 
![image](https://user-images.githubusercontent.com/33631044/109820925-77e07900-7c5b-11eb-9840-cfda799d4034.png)

When I try to make a POST request to backend the connection timesout,
![image](https://user-images.githubusercontent.com/33631044/109821379-ecb3b300-7c5b-11eb-8157-f1bc3d3e89dc.png)

However when I expose the sa-web-app-lb service using `minikube service sa-web-app-lb`, I am able to access it via Postman
![image](https://user-images.githubusercontent.com/33631044/109821696-42885b00-7c5c-11eb-8a96-05d9fdfc596c.png)

What am I doing wrong?
